tools/distpartitioning/data_shuffle.py

```diff
@@ -794,23 +814,25 @@ def read_dataset(rank, world_size, id_lookup, params, schema_map):
         world_size,
         params.num_parts,
         schema_map,
+        ntype_counts,
     )
     logging.info(f"[Rank: {rank}] Done reading dataset {params.input_dir}")
+    dist.barrier()  # SYNCH
```

Comment:
# SYNC?

Reply:
This is put in for the following purpose. 

We now support num_chunks < num_partitions. This means that on some ranks it is quite possible that there is no data to read from (no edge files, no edge/node feature files). In this case, these ranks the disk IO functions (get_dataset) are completed in a jiffy and on other ranks a lot of time is spent in reading files from the disk. 

In these instances, if we want all the ranks to be in sync then we have to barrier() call here. 
Otherwise, debugging  can become really hard.